File: shopping-cart/shopping-cart-java/inventory-api/src/main/java/com/example/inventory/api/InventoryService.java
package com.example.inventory.api;

import akka.Done;
import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.javadsl.api.transport.Method;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.restCall;

/**
 * The inventory serivce interface.
 * <p>
 * This describes everything that Lagom needs to know about how to serve and
 * consume the inventory service.
 */
public interface InventoryService extends Service {
    /**
     * Get the inventory level for the given product id.
     */
    ServiceCall<NotUsed, Integer> get(String productId);

    /**
     * Add inventory to the given product id.
     */
    ServiceCall<Integer, Done> add(String productId);

    @Override
    default Descriptor descriptor() {
        return named("inventory")
                .withCalls(
                        restCall(Method.GET, "/inventory/:productId", this::get),
                        restCall(Method.POST, "/inventory/:productId", this::add)
                )
                .withAutoAcl(true);
    }
}

File: shopping-cart/shopping-cart-scala/inventory-api/src/main/scala/com/example/inventory/api/InventoryService.scala
package com.example.inventory.api

import akka.Done
import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.transport.Method
import com.lightbend.lagom.scaladsl.api.Service
import com.lightbend.lagom.scaladsl.api.ServiceCall

/**
 * The inventory service interface.
 *
 * This describes everything that Lagom needs to know about how to serve and
 * consume the inventory service.
 */
trait InventoryService extends Service {

  /**
   * Get the inventory level for the given item id.
   */
  def get(itemId: String): ServiceCall[NotUsed, Int]

  /**
   * Add inventory to the given item id.
   */
  def add(itemId: String): ServiceCall[Int, Done]

  final override def descriptor = {
    import Service._

    named("inventory")
      .withCalls(
        restCall(Method.GET, "/inventory/:itemId", get _),
        restCall(Method.POST, "/inventory/:itemId", add _)
      )
      .withAutoAcl(true)
  }
}
